packages/authentication/src/types.ts

```diff
@@ -16,6 +16,10 @@ export interface AuthenticateFn {
   (request: Request): Promise<UserProfile | undefined>;
 }
 
+export interface convertUserToUserProfileFn<U> {
```

Comment:
We use PascalCase for type names and usually use a noun, not a verb.

```suggestion
export interface UserToUserProfileConverterFn<U> {
```